### DISCUSSION TITLE
Environment autocompletion issues.

### DISCUSSION THREAD

**Discussion by Lytchak** [texstudio-org/texstudio#2369] [Q&A]
## 2022-5-28
\begin{} \end{} is automatically completed, but if you enter the environment in \begin{}, the environment corresponding to \end{} will not be changed at the same time. Because I clicked close in configuration->completion->Insert Arguments (every time the automatic completion of \frac is \frac{num}{den} It is very troublesome to manually delete the numerator and denominator of the completion, so I closed this option), but I still want to have commands that fill in the \begin and \end environments at the same time, can I?
## 2022-6-6
I want to change the default completion settings (similar to macro definitions). For example, when I type \[, it gives me \[ .\] instead of the default \[\]. Can this be done?

**Comment by sunderme** [comment]
if the arguments are not filled, all other automatisms don't work.
You know about placeholder and jump to next placeholder (usually ctrl+right) ? Then the argument is selected and you can simply overwrite it.

**Comment by Lytchak** [comment]
I have a bold idea: can all the filled parameters be changed to empty or blank?

**Comment by sunderme** [comment]
jumping to placeholders with ctrl+right and filling it in directly has been working for years.
I don't see why it should be changed.
